src/components/MerchantList.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import Merchant from './Merchant';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+
+const initialMerchants = [
+  { id: 0, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 1, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 2, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 3, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 4, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 5, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+];
+
+function MerchantList() {
+  const [merchants, setMerchants] = useState(initialMerchants); 
+  const [hasMore, setHasMore] = useState(true); 
+  const [loading, setLoading] = useState(false); 
+  const [isBottom, setIsBottom] = useState(false);
+  const loadMoreMerchants = () => {
+    if (loading || !hasMore) return;
+    setLoading(true);
+    setTimeout(() => {
+      const newMerchants = merchants.concat(
+        Array.from({ length: 3 }).map((_, index) => ({
+          id: merchants.length + index,
+          name: `新商家 ${merchants.length + index}`,
+          distance: (Math.random() * 10).toFixed(1),
+          costDownLimit: Math.floor(Math.random() * 100),
+          costUpLimit: Math.floor(Math.random() * 200),
+          starRate: (Math.random() * 5).toFixed(1),
+          starNumber: Math.floor(Math.random() * 100),
+        }))
+      );
+
+      setMerchants(newMerchants);
+      
+      if (newMerchants.length >= 50) {
+        setHasMore(false);
+      }
+      setLoading(false); 
+      setIsBottom(false);
+    }, 2000);
+  };
+
+  const handleScroll = () => {
+    const scrollTop = window.pageYOffset || document.documentElement.scrollTop;
+    const windowHeight = window.innerHeight;
+    const documentHeight = document.documentElement.scrollHeight;
+    if (scrollTop + windowHeight >= documentHeight - 30 && !loading && hasMore) {
+      setIsBottom(true); 
+    }
+  };
+
+  useEffect(() => {
+    if (isBottom) loadMoreMerchants();
+  }, [isBottom]);
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, [loading, hasMore]);
+
+  return (
+    <div>
```

Comment:
To add CSS styles so that the Merchant card is displayed in the center of the screen, and if the screen width is wide enough to fit multiple cards, each row should maintain the same number of cards. If the last row has fewer cards, they should still be aligned from the beginning to the end.

Also, when the List shows a scroll bar, it should not affect the header area; the scroll bar should appear below the header.

![image](https://github.com/user-attachments/assets/fd799df8-4d04-4bae-abe9-56a51d570405)